[Outlook](SSO) Add article on SSO in event-based activation (#3074)
OfficeDev/office-js-docs-pr: docs/outlook/use-sso-in-event-based-activation.md

---
title: Enable single sign-on (SSO) in Outlook add-ins that use event-based activation

> description: 'Learn how to enable SSO when working in an event-based activation add-in'

Reviewer: ```suggestion
description: 'Learn how to enable SSO when working in an event-based activation add-in.'
```

ms.date: 11/11/2021
ms.localizationpriority: medium
---

# Enable single sign-on (SSO) in Outlook add-ins that use event-based activation

> When an Outlook add-in uses event-based activation, the events run in a separate JavaScript runtime. After completing the steps in [Authenticate a user with a single-sign-on token in an Outlook add-in](authenticate-a-user-with-an-sso-token.md), this article describes additional steps to enable SSO for your event handling code. Once you enable SSO you can call the `getAccessToken()` API to get an access token with the user's identity.

Reviewer: ```suggestion
When an Outlook add-in uses event-based activation, the events run in a separate JavaScript runtime. After completing the steps in [Authenticate a user with a single-sign-on token in an Outlook add-in](authenticate-a-user-with-an-sso-token.md), follow the additional steps described in this article to enable SSO for your event handling code. Once you enable SSO you can call the `getAccessToken()` API to get an access token with the user's identity.
```

The following example shows how to enable SSO for two add-ins (a main version and beta version). You can list as many add-ins as necessary depending on how many you provide from your web server.

```json
{
    "allowed":
    [
        "https://addin.contoso.com:8000/main/js/autorun.js",
        "https://addin.contoso.com:8000/beta/js/autorun.js"
    ]
}
```

Host the JSON file under a location named `.well-known` in the URI at the root of the origin. For example, if the origin is `https://addin.contoso.com:8000/`, then the well-known URI is `https://addin.contoso.com:8000/.well-known/microsoft-officeaddins-allowed.json`.

The origin refers to a pattern of scheme + subdomain + domain + port. The name of the location **must** be `.well-known`, and the name of the resource file **must** be `microsoft-officeaddins-allowed.json`. This file must contain a JSON object with an attribute named `allowed` whose value is an array of all JavaScript files authorized for SSO for their respective add-ins.

## Add a custom response header

A second approach is to add a custom response header named `MS-OfficeAddins-Allowed-Origin`. The value of the header must be the origin of the JavaScript file.

> For example, if the JavaScript file is located at `https://addin.contoso.com:8000/main/js/autorun.js`, then add the following response header:
> `MS-OfficeAddins-Allowed-Origin : https://addin.contoso.com:8000`

Reviewer: ```suggestion
For example, if the JavaScript file is located at `https://addin.contoso.com:8000/main/js/autorun.js`, then add the following response header.

`MS-OfficeAddins-Allowed-Origin : https://addin.contoso.com:8000`
```
